.ci/Jenkinsfile

```diff
@@ -0,0 +1,206 @@
+pCloud = "colo-research-01"
+gitUrl = null
+gitBranch = null
+gitCommit = null
+pytorchDockerChanged = null
+runWithChecks = null
+expandDockerMatrix = null
+prChangeset = null
+builds = []
+
+def cloneJenkinsfilesRepo() {
+    // Clone the remote jenkins file in WORKSPACE_TMP
+    dir ("$WORKSPACE_TMP") {
+        checkout([  
+            $class: 'GitSCM', 
+            branches: [[name: 'main']], // TODO RJPP_BRANCH
+            doGenerateSubmoduleConfigurations: false, 
+            extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'jenkinsfiles']], 
+            submoduleCfg: [], 
+            userRemoteConfigs: [[url: 'https://github.com/mosaicml/testing', credentialsId: "9cf9add1-2cdd-414b-8160-94bd4ac4a13d"]]  // TODO RJPP_SCM_URL
+        ])
+        return "$WORKSPACE_TMP/jenkinsfiles"
+    }
+}
+
+def runPytest(pythonVersion, extraDeps, gpu = false, marker = "") {
+    def name = "pytest/python${pythonVersion}-extraDeps_${extraDeps}-$marker"
+    def title = "Pytest - Python ${pythonVersion}, composer[${extraDeps}] ($marker)"
+    def summary = title
+    def pytorchVersion = pythonVersion == "3.9" ? "1.10.0" : "1.9.1"
+    def pDockerImage = "mosaicml/pytorch:${pytorchVersion}_cpu-python${pythonVersion}-ubuntu20.04"
+    def pytestCliArgs = '--test_duration all'
+    def nGpus = "0"
+    def memLimit = "7Gi"
+    def cpuLimit = "2"
+
+    if (gpu){
+        nGpus = "2"
+        cpuLimit = "16" // 8 cpu per gpu
+        memLimit = "15Gi"  // 7.5Gb per gpu
+        def cudaVersion = pythonVersion == "3.9" ? "cu113" : "cu111"
+        pDockerImage = "mosaicml/pytorch:${pytorchVersion}_${cudaVersion}-python${pythonVersion}-ubuntu20.04"
+    }
+    if (marker) {
+        pytestCliArgs = "$pytestCliArgs -m $marker"
+    }
+    runWithChecks(
+        name: name,
+        title: title,
+        summary: summary,
+    ) {
+        echo "$title"
+        builds << build(
+            job: 'scratch/pytest',
+            parameters: [
+                string(name: 'P_CLOUD', value: pCloud),
+                string(name: 'P_GIT_REPO', value: gitUrl),
+                string(name: 'P_GIT_COMMIT', value: gitCommit),
+                string(name: 'P_DOCKER_IMAGE', value: pDockerImage),
+                string(name: 'P_CPU_LIMIT', value: cpuLimit),
+                string(name: 'P_MEM_LIMIT', value: memLimit),
+                string(name: 'P_TIMEOUT', value: '1800'),
```

Comment:
TODO - move timeout values to global vars (or something), to make updates easier